core/server/worker/src/main/java/alluxio/worker/netty/DataServerShortCircuitReadHandler.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.netty;
+
+import alluxio.exception.BlockDoesNotExistException;
+import alluxio.exception.InvalidWorkerStateException;
+import alluxio.exception.status.AlluxioStatusException;
+import alluxio.network.protocol.RPCProtoMessage;
+import alluxio.proto.dataserver.Protocol;
+import alluxio.util.proto.ProtoMessage;
+import alluxio.worker.block.BlockLockManager;
+import alluxio.worker.block.BlockWorker;
+
+import com.google.common.base.Preconditions;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Netty handler that handles short circuit read requests.
+ */
+@NotThreadSafe
+class DataServerShortCircuitReadHandler extends ChannelInboundHandlerAdapter {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(DataServerShortCircuitReadHandler.class);
+
+  /** The block worker. */
+  private final BlockWorker mBlockWorker;
+  /** The lock Id of the block being read. */
+  private long mLockId;
+
+  /**
+   * Creates an instance of {@link DataServerShortCircuitReadHandler}.
+   *
+   * @param blockWorker the block worker
+   */
+  DataServerShortCircuitReadHandler(BlockWorker blockWorker) {
+    mBlockWorker = blockWorker;
+    mLockId = BlockLockManager.INVALID_LOCK_ID;
+  }
+
+  @Override
+  public void channelRead(ChannelHandlerContext ctx, Object msg) {
+    if (!acceptMessage(msg)) {
+      ctx.fireChannelRead(msg);
+      return;
+    }
+
+    ProtoMessage message = ((RPCProtoMessage) msg).getMessage();
+    if (message.isLocalBlockOpenRequest()) {
+      handleBlockOpenRequest(ctx, message.asLocalBlockOpenRequest());
+    } else {
```

Comment:
explicitly check for `isLocalBlockCloseRequest()`, and add a last `else` block for error handling.